### PR TITLE
Support building on Github Actions or CircleCI

### PR DESCRIPTION
**Pull request by orangejulius**
We recently moved the Pelias project away from TravisCI, and are now running our unit tests on Github Actions.

Docker image builds, though, are still run on CircleCI. It might be nice to move everything to a single CI provider, so this PR allows the Docker image build script to work on _either_ CircleCI or Github Actions.

### CODE REVIEW

**Review by missinglink — APPROVED** [review]
nice! love the feature

the code is a little hard to follow for someone who didn't write it, would it be possible to abstract out the variables a bit more so it was clear which bit is coming from where?

**Comment by orangejulius** [comment]
Ok, I've improved the code a bit and added many more comments. This should be good to go.

I'd like to record for the world that I'm frustrated GitHub doesn't expose the organization and repository as two separate variables, like CircleCI does. But overall this wasn't too bad.
